simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -664,7 +664,7 @@ Experimental::simd<double, Experimental::simd_abi::avx2_fixed_size<4>> sqrt(
       _mm256_sqrt_pd(static_cast<__m256d>(a)));
 }
 
-#ifdef __INTEL_COMPILER
+#if defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_INTEL_LLVM)
```

Comment:
Maybe, we really leave this for later. I tried playing with it some and can confirm that
- I explicitly need to link with `svml` in a `SYCL+Cuda` build
- I get slightly different results making the tests fail even with `-fp-model=precise` (between 1 ULP and 4 ULP)

In the end, I would expect that we should allow some tolerance for floating point operations as we do in the `KokkosCore` unit test for math functions.